discovery/spec.md

```diff
@@ -365,775 +244,693 @@ The following sections define the attributes that appear in a Service entity.
+  - OPTIONAL
+- Examples:
+  - TBD
 
-##### subscriptionurl
+#### usage
 
-- Type: `URL`
-- Description: An absolute URL indicating where CloudSubscriptions `subscribe`
-  API calls MUST be sent to.
+- Type: String
+- Description: The interaction model supported by this Endpoint. This
+  specification defines a set of possible values (see the
+  [Usage Values](#usage-values) section for more information, however,
+  implementations MAY define new values as extensions. It is RECOMMENDED
+  that extension values be defined with some organizational unique string
+  to reduce the chances of collisions with possible future spec-defined values.
 - Constraints:
   - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - `consumer`
+
+#### config
 
-##### subscriptionconfig
+- Type: Object
+- Description: Metadata the provides additional information concerning the
+  interactions with this Endpoint. Each `usage` value MUST define the set of
+  `config` properties that are valid.
 
-- Type: `Map` of `String` to `String`
-- Description: A map indicating supported options for the `config` parameter for
-  the CloudSubscriptions subscribe() API call. Keys are the name of keys in the
-  allowed config map, the values indicate the type of that parameter, conforming
-  to the CloudEvents
-  [type system](../cloudevents/spec.md#type-system).
-  TODO: Needs resolution with CloudSubscriptions API
+  This property MAY be serialized as an empty object or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
 - Examples:
-  - ??
+  - `{ "protocol": "kafka" , "endpoint": "https://example.com/queue/1234" }`
+
+#### groups
 
-##### subscriptiondialects
+- Type: Array
+- Description: A set of Definition Groups supported by this Endpoint. Each
+  Group MAY be a full representation of the Group or MAY be a
+  [reference](#references) to a Group defined elsewhere. If it is a reference
+  then it MAY be to a Group defined outside of the current Discovery Service
+  instance.
 
-- Type: List of `String` values
-- Description: An array of filter dialects that MAY be used in the
-  Cloud Subscriptions subscribe() API call.
+  This property MAY be serialized as an empty array or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
+  - if present, the same Group, as identified by its `id` or `uri`, MUST NOT
+    appear more than once within this property.
 - Examples:
-  - `basic`
+  - `[ { ...group properties... }, { "uri": "https://example.com/discovery/group/987", "name", "John's blob store messages" } ]`
 
-##### authscope
+#### definitions
 
-- Type: `String`
-- Description: Authorization scope needed for creating subscriptions. The actual
-  meaning of this field is determined on a per-service basis.
+- Type: Array
+- Description: A set of Message Definitions supported by this Endpoint. Each
+  Definition MAY be a full representation of the Definition or MAY be a
```

Comment:
same comments on not wanting to mix full resources and pointers here, I prefer URI-Ref